sink/sink

```diff
@@ -412,17 +417,62 @@ def mkdir_and_chdir(base, identifier):
     os.fchmod(dirfd, 0o755)
     os.close(dirfd)
 
+def prune_logs(base_path = ".", dry_run = False):
+    reference_time = time.time()
+
+    entries = os.listdir(base_path)
+    for entry in entries:
+        path = os.path.join(base_path, entry)
+        log_path = os.path.join(path, "log")
+
+        # we only care about directories with our "log" file in them
+        if not os.path.isdir(path) or not os.path.exists(log_path):
+            continue
+
+        # keep release logs
+        if entry.startswith("release-"):
+            continue
+
+        # use the last modified time of the logfile itself, not just the directory
+        last_modified = os.stat(log_path).st_mtime
+
+        diff_days = (reference_time - last_modified) / (60*60*24)
+
+        # keep anything newer than 30 days old
+        if diff_days < 30:
+            continue
+
+        archive_dir = os.path.join(base_path, "archive")
```

Comment:
Is this meant to delete all attachments and keep the status and log? If so, why don't we just do that? You may have a reason. If so, worth mentioning in the commit log message.